src/Campaigns/Migrations/MigrateFormsToCampaignForms.php

```diff
@@ -179,15 +197,20 @@ protected function addUpgradedV2FormToCampaign($data): void
     }
 
     /**
+     * @unreleased Use the ON DUPLICATE KEY UPDATE statement to prevent exceptions related to duplicate entries
      * @since 4.0.0
      */
     protected function addCampaignFormRelationship($formId, $campaignId)
     {
-        DB::table('give_campaign_forms')
-            ->insert([
-                'form_id' => $formId,
-                'campaign_id' => $campaignId,
-            ]);
+        $table = DB::prefix('give_campaign_forms');
+
+        DB::query(
+            DB::prepare("INSERT INTO {$table} (form_id, campaign_id) VALUES (%d, %d) ON DUPLICATE KEY UPDATE form_id = form_id",
```

Comment:
Hey Glauber, I believe we shouldn't do this. This is a clear indication that our query or logic for migrating forms is wrong.

Reply:
This was my initial idea of a "just in case" solution for situations where duplicated data is added to this table by accident. On the other hand, it can also prevent us from discovering new problems with those queries if they happen. So, I removed it and rolled back the `whereNotIn` statement in the query of the `getUpgradedV2FormsData()` method to ensure we are not retrieving data already added to the `give_campaign_forms` table previously.

Resolved in f387ca46537cbd2bfe71ca5a136bfea4456ccc0f